### PR TITLE
dbt-adapter: add source table materialization

### PR DESCRIPTION
**Pull request by bobbyiliev**
### Motivation

Introducing a new `source_table` materialization as part of the source versioning changes.

Example:

```sql
{{ config(
    materialized='source_table',
    database='materialize'
) }}
FROM SOURCE {{ ref('test_subsources') }}
(REFERENCE "bids")
```

SQL:

```sql
CREATE TABLE "materialize"."public"."source_table_test"
FROM SOURCE "materialize"."public"."test_subsources"
(REFERENCE "bids")
```

Open to some early feedback before adding more tests and updating the documentation!

### CODE REVIEW

File: misc/dbt-materialize/tests/adapter/fixtures.py [hunk 1]
```diff
@@ -96,6 +96,15 @@
 FORMAT BYTES
 """
 
+test_source_table = """
+{{ config(
+    materialized='source_table',
+    database='materialize'
+) }}
+FROM SOURCE {{ ref('test_subsources') }}
+(REFERENCE "bids") WITH (TEXT COLUMNS ["amount"])
```
[the remark this answers is not in the document]
Comment (bobbyiliev) [reply misc/dbt-materialize/tests/adapter/fixtures.py]: Oh interesting, as far as I can tell it seems to be just ignored silently.

[the remark this answers is not in the document]
Comment (morsapaes) [reply misc/dbt-materialize/tests/adapter/fixtures.py]: Sorry, I should've added more context and mention that I manually tested this and was surprised it is silently ignored. Unrelated to the adapter, just wanted to point that out to Roshan!

**Review by morsapaes — APPROVED** [review]
The new materialization type looks good to me! Did some manual tinkering and can't think of anything we need to special case for this (other than blue-green deployments, as a follow-up). This fits dbt's `source`->`table` paradigm so much better, too!

Can we just add a test that uses the `index` config?

**Comment by bobbyiliev** [comment]
> The new materialization type looks good to me! Did some manual tinkering and can't think of anything we need to special case for this (other than blue-green deployments, as a follow-up). This fits dbt's `source`->`table` paradigm so much better, too!
> 
> Can we just add a test that uses the `index` config?

Good point! Just added that test. 

On the documentation side, as we still don't have this documented for the actual functionality on the Materialize side, we could either hold with merging this PR or we could merge it and then have a follow-up PR for the dbt documentation itself.

**Review by morsapaes — APPROVED** [review]

**Pull request merged**